staging/src/k8s.io/apiserver/pkg/server/options/serving.go

```diff
@@ -178,22 +188,18 @@ func (s *SecureServingOptions) ApplyTo(c *server.Config) error {
 	default:
 		c.LoopbackClientConfig = secureLoopbackClientConfig
 		c.SecureServingInfo.SNICerts[server.LoopbackClientServerNameOverride] = &tlsCert
+		c.SecureServingInfo.Listener = s.Listener
 	}
 
 	return nil
 }
 
 func (s *SecureServingOptions) applyServingInfoTo(c *server.Config) error {
-	if s.BindPort <= 0 {
```

Comment:
`listener=0 && BindPort==0` still means to disable secure serving, doesn't it? We should keep supporting that.

Reply:
Yes port 0 still mean do not start server

Reply:
It is guaranteed by `SecureServingOptions.ApplyTo`, here is duplicate, so i remove.

Reply:
makes sense